generator/internal/discovery_type_vertex.cc

```diff
@@ -38,10 +46,13 @@ absl::optional<std::string> CheckForScalarType(nlohmann::json const& j) {
 
 DiscoveryTypeVertex::DiscoveryTypeVertex(
     std::string name, std::string package_name, nlohmann::json json,
-    google::protobuf::DescriptorPool const*)
+    google::protobuf::DescriptorPool const* descriptor_pool)
     : name_(std::move(name)),
       package_name_(std::move(package_name)),
-      json_(std::move(json)) {}
+      json_(std::move(json)),
+      descriptor_pool_(descriptor_pool) {
+  assert(descriptor_pool != nullptr);
```

Comment:
Nit:  FWIW, I'd name the member variable here instead of the parameter when the latter is only used to directly initialize the former.